pkg/minikube/cluster/cluster.go

```diff
@@ -62,6 +64,11 @@ const (
 //INFO lvl logging is displayed due to the kubernetes api calling flag.Set("logtostderr", "true") in its init()
 //see: https://github.com/kubernetes/kubernetes/blob/master/pkg/util/logs/logs.go#L32-L34
 func init() {
+	provision.Register("minishift", &provision.RegisteredProvisioner{
```

Comment:
So how complete is this to be? It still does not address the problem of provider selection, right? Looking at the underlying code, the register call puts the previsioner  into a map keyed against its provided name:

    make(map[string]*RegisteredProvisioner)

Later it just iterates over this map and takes the first provisioner which is compatible with the host  based on the release info.

    for _, p := range provisioners {
    ....
    }

from the Go specs:

> When iterating over a map with a range loop, the iteration order is not specified and is not guaranteed to be the same from one iteration to the next. 

So, if this code is working at all, it is by pure chance. Do we agree?